### DISCUSSION TITLE
Doctype

### DISCUSSION THREAD

**Discussion by dagmanolis** [downforcetech/fluidxml.php#59] [Questions] [answered]
Awesome lib!

Is there a way to write the DOCTYPE definition at the export ?

**Comment by daniele-orlando** [comment, marked as answer]
Hi @dagmanolis. Thanks.

If you have not, give a look a the Wiki page here: [Exporting](https://github.com/servo-php/fluidxml/wiki/Getting-Started:-10.-Exporting).

To export with an HTML 5 doctype, you can use `$xml->html();`
which produces

```xml
<!DOCTYPE html>
<book>
  ...
</book>
```

otherwise you can remove the doctype completely with `$xml->html(true);`
which produces

```xml
<book>
  ...
</book>
```

and you can prepend any doctype you want.

```php
$doc_string = 
    $custom_doctype_string
    . $xml->html(true)
;
```

**Comment by dagmanolis** [comment]
Thanks a lot ! Have a nice day.

On Mon, May 2, 2022 at 8:22 PM Daniele Orlando ***@***.***>
wrote:

> Hi @dagmanolis <https://github.com/dagmanolis>. Thanks.
>
> If you have not, give a look a the Wiki page here: Exporting
> <https://github.com/servo-php/fluidxml/wiki/Getting-Started:-10.-Exporting>
> .
>
> To export with an HTML 5 doctype, you can use $xml->html();
> which produces
>
> <!DOCTYPE html>
> <book>
>   ...
> </book>
>
> otherwise you can remove the doctype completely with $xml->html(true);
> which produces
>
> <book>
>   ...
> </book>
>
> and you can prepend any doctype you want.
>
> $doc_string =
>     $custom_doctype_string
>     . $xml->html(true)
> ;
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/servo-php/fluidxml/discussions/59#discussioncomment-2675300>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKNPO4BUTVI7J2NLCTLYNJDVIAFLVANCNFSM5U2MELBA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Dagdilelis Emmanouil*
+30 6972 129 567

founder: *WebXL - Quality Webing* | https://webxl.gr | blog:
https://manosdag.gr
